tests/test_nodes.py

```diff
@@ -0,0 +1,101 @@
+import os
+
+import httpx
+import pytest
+
+from app.api import utility as util
+
+
+@pytest.mark.parametrize(
+    "local_nodes",
+    [
+        "(https://mylocalnode.org, Local Node)",
+        "(https://mylocalnode.org/, Local Node) (https://firstpublicnode.org/, First Public Node)",
+    ],
+)
+def test_nodes_discovery_endpoint(test_app, monkeypatch, local_nodes):
+    """Test that a federation node index is correctly created from locally set and remote node lists."""
+    monkeypatch.setenv("LOCAL_NB_NODES", local_nodes)
+    monkeypatch.setattr(
+        util,
+        "LOCAL_NODES",
+        os.environ.get(
+            "LOCAL_NB_NODES", "(https://api.neurobagel.org/, OpenNeuro)"
+        ),
```

Comment:
why are you mocking this part as well? i.e. what's the benefit over just directly mocking this as 

```python
monkeypatch.setattr(
        util,
        "LOCAL_NODES",
local_nodes)
```

Reply:
I think you're right that manually setting the environment variable and explicitly reading it in as part of the test is probably more complex then is needed for these integration tests.

Switching to your recommendation!